apps/web/vibes/soul/primitives/skeleton/index.tsx

```diff
@@ -0,0 +1,135 @@
+ *   --skeleton-background: hsl(var(--contrast-100));
+ * }
+ * ```
+ */
+
+export function Skeleton({
+  parentClassName,
+  childClassName,
+  type = 'text',
+  rounded,
+  characterCount = 10,
+  aspectRatio = '5:6',
+}: Props) {
+  if (type === 'box') {
+    return (
+      <div className={clsx('flex items-center', parentClassName)}>
+        <div
+          className={clsx(
+            'bg-[var(--skeleton-background,hsl(var(--contrast-100)))]',
+            typeof rounded === 'string' &&
+              {
+                sm: 'rounded-sm',
+                base: 'rounded',
+                md: 'rounded-md',
+                lg: 'rounded-lg',
+                xl: 'rounded-xl',
+                '2xl': 'rounded-2xl',
+                '3xl': 'rounded-3xl',
+                full: 'rounded-full',
+              }[rounded],
+            childClassName,
+          )}
+        />
+      </div>
+    );
+  }
+
+  if (type === 'circle') {
+    return (
+      <div className={clsx('flex items-center', parentClassName)}>
+        <div
+          className={clsx(
+            'rounded-full bg-[var(--skeleton-background,hsl(var(--contrast-100)))]',
+            childClassName,
+          )}
+        />
+      </div>
+    );
+  }
+
+  if (type === 'image') {
+    return (
+      <div className={clsx('flex items-center', parentClassName)}>
+        <div
+          className={clsx(
+            'bg-[var(--skeleton-background,hsl(var(--contrast-100)))]',
+            typeof rounded === 'string' &&
+              {
+                sm: 'rounded-sm',
+                base: 'rounded',
+                md: 'rounded-md',
+                lg: 'rounded-lg',
+                xl: 'rounded-xl',
+                '2xl': 'rounded-2xl',
+                '3xl': 'rounded-3xl',
+                full: 'rounded-full',
+              }[rounded],
+            {
+              '5:6': 'aspect-[5/6]',
+              '3:4': 'aspect-[3/4]',
+              '1:1': 'aspect-square',
+            }[aspectRatio],
+            childClassName,
+          )}
+        />
+      </div>
+    );
+  }
+
+  return (
+    <div className={clsx('flex h-[1lh] items-center', parentClassName)}>
+      <div
+        className={clsx(
+          'h-[0.5em] bg-[var(--skeleton-background,hsl(var(--contrast-100)))]',
+          typeof rounded === 'string' &&
+            {
+              sm: 'rounded-sm',
+              base: 'rounded',
+              md: 'rounded-md',
+              lg: 'rounded-lg',
+              xl: 'rounded-xl',
+              '2xl': 'rounded-2xl',
+              '3xl': 'rounded-3xl',
+              full: 'rounded-full',
+            }[rounded],
+          childClassName,
+        )}
+        style={{ width: `${characterCount}ch` }}
+      />
+    </div>
+  );
+}
+
+export function SkeletonWrapper({
+  children,
+  className,
+}: {
+  children: React.ReactNode;
+  className?: string;
+}) {
+  return (
+    <div className={clsx('motion-safe:animate-pulse', className)} role="status">
```

Comment:
I think that the issue with adding the pulse animation here is that we won't be able to change whether we pulse or not based on whether we're building an empty state or not. When I think of a "skeleton" it could be used both for loading or empty states, but whether it's pulsing depends on the pending state. Also, for situations like the PLP, the pending state is based on a sibling so we still want to use Tailwind `group` for that.

I like that we're encapsulating the accessibility, etc. but perhaps this component should be named `PendingWrapper` or something else? Also, do we want to have this sort of stuff in PLP? Do we need to go back to context so we can use `aria-hidden` and `sr-only` or is there a way we can do this with just CSS?